wdl-lint/src/rules/command_mixed_indentation.rs

```diff
@@ -115,83 +143,161 @@ impl Visitor for CommandSectionMixedIndentationRule {
-                                        // Mixed indentation, store the span of the first mixed
-                                        // character
-                                        mixed_span =
-                                            Some(Span::new(text.span().start() + start + i, 1));
-                                        break 'outer;
+                                // Otherwise, check the leading whitespace
+                                for (i, b) in line.as_bytes().iter().enumerate() {
+                                    match b {
+                                        b' ' | b'\t' => {
+                                            let current = IndentationKind::from(*b);
+                                            let kind = kind.get_or_insert(current);
+                                            if current != *kind {
+                                                // Mixed indentation, store the span of the first
+                                                // mixed
+                                                // character
+                                                mixed_span = Some(Span::new(
+                                                    text.span().start() + start + i,
+                                                    1,
+                                                ));
+                                                break 'outer;
+                                            }
+                                        }
+                                        _ => break,
                                     }
                                 }
-                                _ => break,
                             }
                         }
+                        CommandPart::Placeholder(_) => {
+                            // Encountered a placeholder, skip the next line of text as it's
+                            // really a part of the same line
+                            skip_next_line = true;
+                        }
                     }
                 }
-                CommandPart::Placeholder(_) => {
-                    // Encountered a placeholder, skip the next line of text as it's
-                    // really a part of the same line
-                    skip_next_line = true;
+
+                if let Some(span) = mixed_span {
+                    let command_keyword =
+                        support::token(section.inner(), SyntaxKind::CommandKeyword)
+                            .expect("should have a command keyword token");
+
+                    state.add(mixed_command_indentation(
+                        command_keyword.text_range().into(),
+                        span,
+                        kind.expect("an indentation kind should be present"),
+                    ));
                 }
             }
+            VisitReason::Exit => {
+                self.in_command_section = false;
+            }
```

Comment:
Instead of embedding all of the logic in the match arm, this should be:
```suggestion
            VisitReason::Exit => {
                self.in_command_section = false;
                return;
            }
```

Then the function logic can move out of the match arm. This should also minimize the diff in the PR review.